### PR TITLE
Add MPRIS Ratings extension

### PR DESCRIPTION
**Pull request by JasonLG1979**
Allows MPRIS clients to love, ban, set tired and unrate songs.
This also deprecates and removes the legacy _dbus_service.py, and refactors the mpris plugin accordingly.

### CODE REVIEW

File: pithos/plugins/mpris.py [hunk 1]
```diff
@@ -12,10 +13,25 @@
...
+#
+# See <https://specifications.freedesktop.org/mpris-spec/latest/interfaces.html>
+# for documentation.
 
+import codecs
 import logging
+import math
 
-from gi.repository import Gio, GLib
+from gi.repository import (
```
Comment (TingPing) [inline pithos/plugins/mpris.py]: __[flake8]__
* [F401] 'Gtk' imported but unused

File: pithos/plugins/mpris.py [hunk 2]
```diff
@@ -25,29 +40,782 @@ class MprisPlugin(PithosPlugin):
...
+
+    @dbus_property(MEDIA_PLAYER2_RATINGS_IFACE, signature='b')
+    def CanRate(self):
+        '''b Read only Interface MediaPlayer2.ExtensionPithosRatings'''
+        # This property exists so that applets can check it to make sure
+        # the MediaPlayer2.ExtensionPithosRatings interface actually exists.
+        # It's much more convenient for them then wrapping all their
+        # ratings code in the equivalent of a try except block.
+        # Not all versions of Pithos will have this interface.
+        # It serves a similar function as HasTrackList.
```
Comment (TingPing) [inline pithos/plugins/mpris.py]: __[flake8]__
* [W291] trailing whitespace

**Comment by JasonLG1979** [comment]
Thoughts on this @TingPing?

**Comment by TingPing** [comment]
Can you break up the commits to it is more readable. First add rating support then remove old interface.

**Comment by JasonLG1979** [comment]
Sure.

**Comment by JasonLG1979** [comment]
That work for you @TingPing?

**Comment by JasonLG1979** [comment]
This one too and then release!!! ;)

File: pithos/plugins/_mpris.py [hunk 3]
```diff
@@ -576,6 +581,17 @@ def CanEditTracks(self):
         '''b Read only Interface MediaPlayer2.TrackList'''
         return False
 
+    @dbus_property(MEDIA_PLAYER2_RATINGS_IFACE, signature='b')
+    def CanRate(self):
```
Comment (TingPing) [inline pithos/plugins/_mpris.py]: This seems very specific to Pithos so maybe this should have a more specific name?

Comment (JasonLG1979) [inline pithos/plugins/_mpris.py]: Can do.

Comment (JasonLG1979) [inline pithos/plugins/_mpris.py]: How about PithosCanRate?

Comment (JasonLG1979) [inline pithos/plugins/_mpris.py]: Really though it's on it's own interface named "ExtensionPithosRatings" I don't think anyone's going to have an interface named that? A client wouldn't even see that prop unless they were specifically trying to use that interface.

Comment (JasonLG1979) [inline pithos/plugins/_mpris.py]: Here's what the interface looks like in Gjs.
https://gist.github.com/JasonLG1979/77c6f03cc1bb23fabc78394f4be31332

Comment (JasonLG1979) [inline pithos/plugins/_mpris.py]: @TingPing what do you suggest?

Comment (TingPing) [inline pithos/plugins/_mpris.py]: Honestly I'd do something like: `HasPithosExtension()`

Comment (JasonLG1979) [inline pithos/plugins/_mpris.py]: Done.

**Comment by JasonLG1979** [comment]
Merge me. I changed what you wanted.

**Pull request merged**